src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewNode.cs

```diff
@@ -0,0 +1,274 @@
+		{
+			var collectionChange = args.CollectionChange;
+			var index = args.Index;
+			UpdateHasChildren();
+			RaiseChildrenChanged(collectionChange, index);
+		}
+
+		private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			DependencyProperty property = args.Property;
+			//TODO:
+			//m_propertyChangedEventSource(this, args);
+		}
+
+		private void RaiseChildrenChanged(CollectionChange collectionChange, uint index)
+		{
+			var args = new VectorChangedEventArgs(collectionChange, index);
+			ChildrenChanged?.Invoke(this, args);
+		}
+
+		internal object ItemsSource
+		{
+			get => m_itemsSource;
+			set
+			{
+				//TODO:
+				//m_itemItemsSourceViewChangedRevoker.revoke();
+				m_itemsSource = value;
+				//m_itemsDataSource = value != null ? new ItemsSourceView(value) : null;
+				//if (m_itemsDataSource)
+				//{
+				//	m_itemItemsSourceViewChangedRevoker = m_itemsDataSource.CollectionChanged(auto_revoke, { this, &TreeViewNode.OnItemsSourceChanged });
+				//}
+				//SyncChildrenNodesWithItemsSource();
+			}
+		}
+
+		private void OnItemsSourceChanged(object sender, NotifyCollectionChangedEventArgs args)
+		{
+			switch (args.Action)
+			{
+				case NotifyCollectionChangedAction.Add:
+					{
+						// TreeViewNode and ItemsSource will update each other when data changes.
+						// For ItemsSource -> TreeViewNode changes, m_itemsDataSource.Count() > Children().Size()
+						// We'll add the new node to children collection.
+						// For TreeViewNode -> ItemsSource changes, m_itemsDataSource.Count() == Children().Size()
+						// the node is already in children collection, we don't want to update TreeViewNode again here.
+						if (m_itemsDataSource.Count != Children.Count))
+						{
+							AddToChildrenNodes(args.NewStartingIndex(), args.NewItems().Size());
+						}
+						break;
+					}
+
+				case NotifyCollectionChangedAction.Remove:
+					{
+						// TreeViewNode and ItemsSource will update each other when data changes.
+						// For ItemsSource -> TreeViewNode changes, m_itemsDataSource.Count() < Children().Size()
+						// We'll remove the node from children collection.
+						// For TreeViewNode -> ItemsSource changes, m_itemsDataSource.Count() == Children().Size()
+						// the node is already removed, we don't want to update TreeViewNode again here.
+						if (m_itemsDataSource.Count() != static_cast<int>(Children().Size()))
+						{
+							RemoveFromChildrenNodes(args.OldStartingIndex(), args.OldItems().Size());
+						}
+						break;
+					}
+
+				case NotifyCollectionChangedAction.Reset:
+					{
+						SyncChildrenNodesWithItemsSource();
+						break;
+					}
+
+				case NotifyCollectionChangedAction.Replace:
+					{
+						RemoveFromChildrenNodes(args.OldStartingIndex(), args.OldItems().Size());
+						AddToChildrenNodes(args.NewStartingIndex(), args.NewItems().Size());
+						break;
+					}
+			}
+		}
+
+		//	void TreeViewNode.AddToChildrenNodes(int index, int count)
+		//	{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)